CRM/Core/Smarty/plugins/function.defaultCheck.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @author Andrew Hunt, AGH Strategies
+ * $Id$
+ *
+ */
+
+/**
+ * Display a checkmark to indicate the default item.
+ *
+ * This is a wrapper around CRM_Core_Page::getDefaultCheckMarkup().
+ *
+ * @param $params
+ *   - condition: if present and falsey, return empty
+ *   - text: the *untranslated* text to display instead of "Default"
+ *   - icon: the icon class to display instead of fa-check
+ *
+ * @param $smarty
+ *
+ * @return string
+ */
+function smarty_function_defaultCheck($params, &$smarty) {
+  if (array_key_exists('condition', $params) && !$params['condition']) {
+    return '';
+  }
+  $text = ts(CRM_Utils_Array::value('text', $params, 'Default'));
+  $icon = CRM_Utils_Array::value('icon', $params, 'fa-check');
```

Comment:
Is there any reason this smarty function can't just be `{icon}` and work for any fa-icon?
Also, please use the coalesce operator rather than `CRM_Utils_Array::value` which I am trying hard to eradicate from our codebase.

```suggestion
  $icon = $params['icon'] ?? 'fa-check';
```